src/microsim/schema/simulation.py

```diff
@@ -93,6 +93,28 @@ def _resolve_spaces(self) -> "Self":
             self.output_space.reference = self.truth_space
         return self
 
+    @model_validator(mode="after")
+    def _check_fluorophores_equal_in_samples(self) -> "Self":
+        fp_names = [{lbl.fluorophore.name for lbl in s.labels} for s in self.samples]
+        if len({frozenset(s) for s in fp_names}) != 1:
+            raise ValueError(
+                "All samples in the batch must use the same set of fluorophores."
+            )
```

Comment:
curious about this.  I can see that this would be necessary in the case where we assume that the sample "axis" is just an extension of a non-jagged array.  But, maybe sample should be special in that case, and maybe samples should always be iterated over in a for-loop rather than processed in a vectorized fashion?  (might help with memory bloat too)

that can be decided later. It's definitely easier to start with this restriction and relax it later.